src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_gssapi.h"
+#include <assert.h>
+#include <string.h>
+
+static_assert(PAL_GSS_COMPLETE == GSS_S_COMPLETE, "");
+static_assert(PAL_GSS_CONTINUE_NEEDED == GSS_S_CONTINUE_NEEDED, "");
+
+static char gss_mech_value[] = "\x2b\x06\x01\x05\x05\x02";
+
+static gss_OID_desc gss_mech_spnego_OID_desc = {6, static_cast<void*>(gss_mech_value)};
+
+static gss_OID gss_mech_spnego_OID = &gss_mech_spnego_OID_desc;
+
+static gss_OID_set_desc gss_mech_spnego_OID_set_desc = {1, &gss_mech_spnego_OID_desc};
+
+static gss_OID_set gss_mech_spnego_OID_set = &gss_mech_spnego_OID_set_desc;
+
+extern "C" uint32_t GssAcquireCredSpNego(uint32_t* minorStatus, gss_name_t desiredName, bool isInitiate, gss_cred_id_t* outputCredHandle)
+{
+    gss_cred_usage_t credUsage = isInitiate ? GSS_C_INITIATE : GSS_C_ACCEPT;
+    return gss_acquire_cred(minorStatus, desiredName, 0, gss_mech_spnego_OID_set, credUsage, outputCredHandle, NULL, NULL);
+}
+
+extern "C" uint32_t GssDeleteSecContext(uint32_t* minorStatus, gss_ctx_id_t* contextHandle)
+{
+    return gss_delete_sec_context(minorStatus, contextHandle, GSS_C_NO_BUFFER);
+}
+
+extern "C" uint32_t GssDisplayName(uint32_t* minorStatus, gss_name_t inputName, gss_buffer_t outputNameBuffer)
+{
+    return gss_display_name(minorStatus, inputName, outputNameBuffer, NULL);
+}
+
+extern "C" uint32_t GssDisplayStatus(uint32_t* minorStatus, uint32_t statusValue, bool isGssMechCode, gss_buffer_t statusString)
+{
+    int statusType = isGssMechCode ? GSS_C_MECH_CODE : GSS_C_GSS_CODE;
+    return gss_display_status(minorStatus, statusValue, statusType, GSS_C_NO_OID, NULL, statusString);
+}
+
+extern "C" uint32_t GssImportName(uint32_t* minorStatus, char* inputName, bool isUser, gss_name_t* outputName)
+{
+    gss_buffer_desc inputNameBuffer {strlen(inputName), inputName};
+    gss_OID nameType = isUser ? const_cast<gss_OID>(GSS_C_NT_USER_NAME) : const_cast<gss_OID>(GSS_KRB5_NT_PRINCIPAL_NAME);
+    return gss_import_name(minorStatus, &inputNameBuffer, nameType, outputName);
+}
+
+extern "C" uint32_t GssInitSecContext(uint32_t* minorStatus, gss_cred_id_t claimantCredHandle, gss_ctx_id_t* contextHandle,
```

Comment:
Similar to GssImportName, I'm wondering if the boolean parameter should be gotten rid of, and this broken into two separate (visible) methods.  (They can have a unified backing implementation into a non-public method)